vec4state.cpp

```
@@ -1,1029 +0,0 @@
-            return vec4state("0", 1);
-        }
-    }
-    for (long long i = 0; i < xorVector.vectorSize; i++) {
-        if (xorVector.vector[i].getBval() != 0) {
-            return vec4state("x", 1);
-        }
-    }
-    return vec4state("1", 1);
-}
-
-vec4state vec4state::operator!=(const vec4state& other) const {
-    return !(*this == other);
-}
-
-vec4state vec4state::caseEquality(const vec4state& other) const {
-    for (int i = 0; i < min(vectorSize, other.vectorSize); i++) {
-        if (vector[i].getAval() != other.vector[i].getAval() || vector[i].getBval() != other.vector[i].getBval()) {
-            return vec4state("0", 1);
-        }
-    }
-    // If the vectors are of different sizes, check if the larger vector has any bits set to 1
-    if (vectorSize > other.vectorSize) {
-        for (long long i = other.vectorSize; i < vectorSize; i++) {
-            if (vector[i].getAval() != 0 || vector[i].getBval() != 0) {
-                return vec4state("0", 1);
-            }
-        }
-    } else if (vectorSize < other.vectorSize) {
-        for (long long i = vectorSize; i < other.vectorSize; i++) {
-            if (other.vector[i].getAval() != 0 || other.vector[i].getBval() != 0) {
-                return vec4state("0", 1);
-            }
-        }
-    }
-    return vec4state("1", 1);
-}
-
-vec4state vec4state::caseInequality(const vec4state& other) const {
-    return !caseEquality(other);
-}
-
-vec4state vec4state::operator<<(const vec4state& other) {
-    if (other.unknown) {
-        return vec4state("x", numBits);
-    }
-    for (long long i = other.vectorSize - 1; i >= 2; i--) {
-        if (other.vector[i].getAval() != 0) {
-            return vec4state("0", numBits);
-        }
-    }
-    long long num = 0;
-    for (long long i = min(other.vectorSize - 1, (long long)(1)); i >= 0; i--) {
-        num = num << 32;
-        num += other.vector[i].getAval();
-    }
-    if (num < 0) {
-        return vec4state("0", numBits);
```

If you define num to be unsigned you won't have this issue.
Otherwise throw an appropriate exception.

---

In Verilog, if num is negative then it returns a vector that is all 0's. Should we throw an exception anyway?